OpenSim/Sandbox/HumphreysImplicitDynamics/forwardImplicitTest.cpp

```diff
@@ -0,0 +1,368 @@
+ *                                                                            *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    *
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         *
+ *                                                                            *
+ * Unless required by applicable law or agreed to in writing, software        *
+ * distributed under the License is distributed on an "AS IS" BASIS,          *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        *
+ * limitations under the License.                                             *
+ * -------------------------------------------------------------------------- */
+
+/* 
+ *  Below is an example of an OpenSim application that provides its own 
+ *  main() routine.  
+ */
+
+//==============================================================================
+//==============================================================================
+#include <OpenSim/OpenSim.h>
+
+using namespace OpenSim;
+using namespace SimTK;
+using namespace std;
+
+struct momentArmAndLength {
+    SimTK::Vec2 length;
+    SimTK::Vec2 momentArm;
+};
+
+
+
+
+struct bodyResidual {
+    double forceResidual = SimTK::NaN;
+    double velocityResidual = SimTK::NaN;
+    double forceMass = SimTK::NaN;
+    SimTK::Mat22 df_dy = {SimTK::NaN,SimTK::NaN,SimTK::NaN,SimTK::NaN};
+    SimTK::Mat22 df_dydot = {SimTK::NaN,SimTK::NaN,SimTK::NaN,SimTK::NaN};
+};
+
+
+
+
+// Get the Residual of the mass body.
+bodyResidual calcBodyImplicitResidual(Model& osimModel, State& s, double velocityGuess, double accelGuess, Vec2 muscleForces)
+{
+
+    // Tried to use new implicit interface:
+    //   (going to just use equation implementation for now)
+    //const double qDotGuess = 0.0; const double uDotGuess = 0.0;
+    // Set derivative guess.
+    //coord.setStateVariableDerivativeGuess(s, "value", qDotGuess);
+    //coord.setStateVariableDerivativeGuess(s, "speed", uDotGuess);
+
+    /*Returns:
+    /home/humphreysb/vanDenBogertMuscleTestSource/vanDenBogertMuscleTest.cpp:220:43: error: no viable conversion from 'SimTK::State' to
+    'const std::string' (aka 'const basic_string<char>')
+    coord.setStateVariableDerivativeGuess(s, "value", qDotGuess);
+     */
+
+    //  Get the "Rigid Body" Dynamics
+
+    Vec2 momentArms={-1, 1};    //Moment Arms are hard coded here.
+                                //Ajay suggests trying to see cost having
+                                //opensim calculate them.
+
+    //Calculate the mass force residual (for now by equations)
+    double m=20;
+    double forceMass = m*accelGuess;  //F=ma
+    double forceRes = momentArms[0]*muscleForces[0] + momentArms[1]*muscleForces[1] - forceMass;
```

Comment:
We definitely want to make sure that the VandenBogert2011Muscle is actually applying forces, so this should eventually use maybe the InverseDynamicsSolver.